tests/basic/storage_pool_test.go

```diff
@@ -9761,3 +9761,192 @@ var _ = Describe("{AddDriveMetadataPool}", func() {
+					} else {
+						err := errors.New("failed while getting msg")
+						log.FailOnError(err, "failed while getting msg")
+					}
+				} else {
+					dash.VerifyFatal(resizeErr != nil, false, fmt.Sprintf("Expected new size to be '%d' or '%d'", expectedSize, expectedSize-3))
+				}
+			}
+			err = Inst().V.RefreshDriverEndpoints()
+			log.FailOnError(err, "error refreshing volume endpoints")
+		})
+
+		Step("Test beyond max pool count", func() {
+			log.InfoD(stepLog)
+			poolList, err := GetPoolsDetailsOnNode(selectedNode)
+			poolLeft := maxPoolLength - len(poolList)
+			err = addNewPools(selectedNode, poolLeft)
+			log.FailOnError(err, "error adding new pools")
+			err = addNewPools(selectedNode, 1)
+			dash.VerifyFatal(err != nil, true, "Error expected as pools trying to create is more than allowed")
+			poolList, err = GetPoolsDetailsOnNode(selectedNode)
+		})
+
+		Step("Test beyond max drives per node", func() {
+			log.InfoD(stepLog)
+			var selectedPools string
+			poolList, err := GetPoolsDetailsOnNode(selectedNode)
+			log.FailOnError(err, "failed while getting updated node pool list")
+			//first add all the new pools
+			poolLeft := maxPoolLength - len(poolList)
+			err = addNewPools(selectedNode, poolLeft)
+			drvNum, err := Inst().N.RunCommand(selectedNode, "lsblk -l -d -e 11 -n -o NAME|wc -l", node.ConnectionOpts{
+				Timeout:         defaultCommandTimeout,
+				TimeBeforeRetry: defaultCommandRetry,
+				IgnoreError:     false,
+				Sudo:            false,
+			})
+			drvNum1 := strings.TrimSpace(string(drvNum))
+			drvNum1 = strings.Trim(drvNum1, "\n")
+			drvNumInt, err := strconv.Atoi(drvNum1)
+			log.FailOnError(err, "failed to convert to int")
+			maxDriveLimit, err := GetPoolMaxCloudDriveLimit(&selectedNode)
+			maxDriveLimitInt := int(maxDriveLimit)
+			maxDriveAllowed := maxDriveLimitInt - drvNumInt
+			err = Inst().V.RefreshDriverEndpoints()
+			log.FailOnError(err, "error refreshing volume endpoints")
+			//expand by adding disk
+			selectedNodeForOps, err := node.GetNodeByName(selectedNode.Name)
+			log.FailOnError(err, "failed while getting node")
+			poolListForOps, err := GetPoolsDetailsOnNode(selectedNodeForOps)
+			log.FailOnError(err, "failed while getting updated node pool list")
+
+			for maxDriveAllowed > 0 {
+				for i := 0; i < len(poolListForOps); i++ {
+					selectedPool, err := GetPoolWithIOsInGivenNode(selectedNode, contexts)
+					log.FailOnError(err, "error in getting pool with IO with error")
+					if selectedPool.Uuid != poolListForOps[i].Uuid {
+						drvSize, err := getPoolDiskSize(poolListForOps[i])
+						log.FailOnError(err, "error getting drive size for pool [%s]", poolListForOps[i].Uuid)
+						//adding 4 drives to each pool which is having one drive.so that max drive per node is accomdated
+						for j := 1; j <= 4; j++ {
```

Comment:
what pool already has max drives allowed? is that condition handled?

Reply:
handled with new code